src/daft-recordbatch/src/ops/window.rs

```diff
@@ -75,4 +75,54 @@ impl RecordBatch {
         // Union the original data with the window result
         self.union(&window_result)
     }
+
+    /// Simplified version of window_agg_sorted that works with pre-sorted single partitions
+    pub fn window_agg_sorted_partition(
```

Comment:
Let's name this one `window_agg` and the original one `window_grouped_agg`. That reflects the behavior of each better. Also we don't use the `group_by` argument in this function, let's remove that